docs/examples/plot_intermediate_outputs.py

```diff
@@ -0,0 +1,268 @@
+There is one change because
+`ONNX-ML Imputer <https://github.com/onnx/onnx/blob/master/docs/Operators-ml.md#ai.onnx.ml.Imputer>`_
+does not handle string type. This cannot be part of the final ONNX pipeline
+and must be removed. Look for comment starting with ``---`` below.
+"""
+import pandas as pd
+import numpy as np
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import StandardScaler, OneHotEncoder
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+titanic_url = ('https://raw.githubusercontent.com/amueller/'
+               'scipy-2017-sklearn/091d371/notebooks/datasets/titanic3.csv')
+data = pd.read_csv(titanic_url)
+X = data.drop('survived', axis=1)
+y = data['survived']
+
+# SimpleImputer on string is not available for string in ONNX-ML specifications.
+# So we do it beforehand.
+for cat in ['embarked', 'sex', 'pclass']:
+    X[cat].fillna('missing', inplace=True)
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+
+numeric_features = ['age', 'fare']
+numeric_transformer = Pipeline(steps=[
+    ('imputer', SimpleImputer(strategy='median')),
+    ('scaler', StandardScaler())])
+
+categorical_features = ['embarked', 'sex', 'pclass']
+categorical_transformer = Pipeline(steps=[
+    # --- SimpleImputer is not available for strings in ONNX-ML specifications. 
+    # ('imputer', SimpleImputer(strategy='constant', fill_value='missing')),
+    ('onehot', OneHotEncoder(handle_unknown='ignore'))])
+
+preprocessor = ColumnTransformer(
+    transformers=[
+        ('num', numeric_transformer, numeric_features),
+        ('cat', categorical_transformer, categorical_features),
+        ])
+
+clf = Pipeline(steps=[('preprocessor', preprocessor),
+                      ('classifier', LogisticRegression(solver='lbfgs'))])
+
+
+clf.fit(X_train, y_train)
+
+##################################
+# Define the inputs of the ONNX graph
+# +++++++++++++++++++++++++++++++++++
+#
+# *sklearn-onnx* does not know the features used to train the model
+# but it needs to know which feature has which name.
+# We simply reuse the dataframe column definition.
+print(X_train.dtypes)
+
+#########################
+# After conversion.
+from skl2onnx.common.data_types import FloatTensorType, StringTensorType, Int64TensorType
+
+def convert_dataframe_schema(df, drop=None):
+    inputs = []
+    for k, v in zip(df.columns, df.dtypes):
+        if drop is not None and k in drop:
+            continue
+        if v == 'int64':
+            t = Int64TensorType([1, 1])
+        elif v == 'float64':
+            t = FloatTensorType([1, 1])
+        else:
+            t = StringTensorType([1, 1])
+        inputs.append((k, t))
+    return inputs
+    
+inputs = convert_dataframe_schema(X_train)
+
+import pprint
```

Comment:
Imports should be at the beginning of a file. Please make changes at other places as well.

Reply:
I usually write examples differently as I write code. I import module for the first time they are needed in the example.